src/Data/BoardReducer.js

```diff
@@ -1,46 +1,75 @@
-import { uuid } from "uuidv4";
-const CREATE_BOARD = "CREATE_BOARD";
+import { api } from "../Api/Api";
+const CREATED_BOARD = "CREATED_BOARD";
 const DELETE_BOARD = "DELETE_BOARD";
+const SET_BOARDS = "SET_BOARDS";
+const CLEAR_DATA = "CLEAR_DATA";
+// let api={}
 
-let localStorage = JSON.parse(window.localStorage.getItem("persist:root"));
-let initialState = localStorage ? localStorage.boards : [];
-const BoardReduser = (state = initialState, action) => {
+// let initialState = [];
+const BoardReduser = (state = [], action) => {
   switch (action.type) {
-    case CREATE_BOARD: {
+    case CREATED_BOARD: {
+      let newState = [...state];
+      newState.push({ id: action.id, name: action.name });
+      return newState;
+    }
+    case SET_BOARDS: {
+      let newState = [...action.state];
+      return newState;
+    }
+    case CLEAR_DATA: {
       let newState = [...action.state];
-      newState.push({ id: uuid(), name: action.name });
       return newState;
     }
     case DELETE_BOARD: {
-      let newState = action.stateBoard.filter(
-        (item) => action.boardId !== item.id
-      );
+      let newState = [...state];
+      newState = newState.filter((item) => action.boardId !== item.id);
       return newState;
     }
     default:
       return state;
   }
 };
 
-export const createBoard = (state, name) => {
-  return { type: CREATE_BOARD, state, name };
+export const createdBoard = ({ id, name }) => {
+  return { type: CREATED_BOARD, id, name };
+};
+export const settedBoards = (state) => {
+  return { type: SET_BOARDS, state };
 };
 
-export const deleteBoard = (
-  stateBoard,
-  boardId,
-  listsId,
-  stateList,
-  stateTask
-) => {
-  return {
-    type: DELETE_BOARD,
-    stateBoard,
-    boardId,
-    listsId,
-    stateList,
-    stateTask,
-  };
+export const deletedBoard = ({ boardId, listsId }) => {
```

Comment:
Please move all actions and thunks to their separate corresponding (boardActions, listActions) files